cpp/new_translation_items_alert.cc

```diff
@@ -0,0 +1,233 @@
+    DbResultSet keywords_new_result_set(ExecSqlAndReturnResultsOrDie(keywords_new_items_query, &db_connection));
+
+    std::vector<std::string> keywords_new_items;
+    while (const auto db_row = keywords_new_result_set.getNextRow())
+        keywords_new_items.emplace_back(db_row["translation"]);
+    names_to_values_map->insertArray("keywords_new_items", keywords_new_items);
+    names_to_values_map->insertScalar("last_notified", last_notified.substr(0, __builtin_strlen("0000-00-00")));
+    return (not(vufind_new_items.empty() && keywords_new_items.empty()));
+}
+
+
+void MailNewItems(const std::string &user, const IniFile &ini_file, const Template::Map &names_to_values_map, const bool debug = false) {
+    std::stringstream mail_content;
+    std::ifstream new_translation_items_template(UBTools::GetTuelibPath() + "translate_chainer/new_translation_items_alert.msg");
+    Template::ExpandTemplate(new_translation_items_template, mail_content, names_to_values_map);
+    const std::string recipient(ini_file.getString(EMAIL_SECTION, user, ""));
+    if (recipient.empty())
+        LOG_ERROR("Could not determine Email address for user \"" + user + "\" section \"" + EMAIL_SECTION + "\" in Ini file \""
+                  + ini_file.getFilename());
+    if (debug) {
+        std::cerr << "CONTENT:" << mail_content.str();
+        return;
+    }
+
+    if (unlikely(not EmailSender::SimplerSendEmail("no-reply@ub.uni-tuebingen.de", { recipient }, "New Translation Items",
+                                                   mail_content.str(), EmailSender::DO_NOT_SET_PRIORITY, EmailSender::HTML)))
+        LOG_ERROR("Could not send mail");
+}
+
+std::string GetNotifyInterval(const IniFile &ini_file, const std::string &user) {
+    std::string notify_interval;
+    ini_file.lookup(NEW_ITEM_NOTIFICATION_SECTION, user, &notify_interval);
+    return notify_interval;
+}
+
+
+std::string GetNotifyThreshold(const IniFile &ini_file, DbConnection &db_connection, const std::string &user) {
+    const std::string notify_interval(GetNotifyInterval(ini_file, user));
+    if (notify_interval.empty())
+        return "";
+    return GetAbsoluteTimeForInterval(db_connection, notify_interval);
+}
+
+
+bool NotifyTimeExceeded(DbConnection &db_connection, const std::string &last_notified, const std::string notify_threshold) {
+    const std::string query("SELECT DATEDIFF('" + notify_threshold + "','" + last_notified + "') <= 0 AS notify_time_exceeded");
+    DbResultSet result(ExecSqlAndReturnResultsOrDie(query, &db_connection));
+    return (bool)std::atoi(result.getNextRow()["notify_time_exceeded"].c_str());
+}
+
+
+void UpdateLastNotifiedTo(DbConnection * const db_connection, const std::string &user, const std::string new_last_notified,
+                          const bool debug = false) {
+    if (debug)
+        return;
+
+    const std::string update_statement("UPDATE translators SET last_notified ='" + new_last_notified + "' WHERE translator='" + user + "'");
```

Comment:
If there are just a few couple of seconds between notification & DB update, we could also pass NOW() here without quotes to save another query if that is exact enough. But since you already implemented it maybe we just leave it as-is.